sdk/python/feast/infra/aws_dynamodb_provider.py

```diff
@@ -0,0 +1,232 @@
+
+import boto3
+import mmh3
+import pandas
+from botocore.exceptions import ClientError
+from tqdm import tqdm
+
+from feast import FeatureTable, utils
+from feast.entity import Entity
+from feast.feature_view import FeatureView
+from feast.infra.key_encoding_utils import serialize_entity_key
+from feast.infra.offline_stores.helpers import get_offline_store_from_sources
+from feast.infra.provider import (
+    Provider,
+    RetrievalJob,
+    _convert_arrow_to_proto,
+    _get_column_names,
+    _run_field_mapping,
+)
+from feast.protos.feast.types.EntityKey_pb2 import EntityKey as EntityKeyProto
+from feast.protos.feast.types.Value_pb2 import Value as ValueProto
+from feast.registry import Registry
+from feast.repo_config import DynamoDbOnlineStoreConfig, RepoConfig
+
+
+class AwsDynamodbProvider(Provider):
+    _wcu: int
+    _rcu: int
+
+    def __init__(self, config: RepoConfig):
+        assert isinstance(config.online_store, DynamoDbOnlineStoreConfig)
+        if config and config.online_store and config.online_store.rcu:
+            self._rcu = config.online_store.rcu
+        else:
+            self._rcu = 5
+
+        if config and config.online_store and config.online_store.wcu:
+            self._wcu = config.online_store.wcu
+        else:
+            self._wcu = 5
+
+        if config and config.online_store and config.online_store.region_name:
+            self.region_name = config.online_store.region_name
+            os.environ["AWS_DEFAULT_REGION"] = self.region_name
+        else:
+            self.region_name = os.environ.get("AWS_DEFAULT_REGION", "us-west-2")
+            os.environ["AWS_DEFAULT_REGION"] = self.region_name
+
+    def _initialize_dynamodb(self):
+        return boto3.resource("dynamodb", region_name=self.region_name)
+
+    def update_infra(
+        self,
+        project: str,
+        tables_to_delete: Sequence[Union[FeatureTable, FeatureView]],
+        tables_to_keep: Sequence[Union[FeatureTable, FeatureView]],
+        entities_to_delete: Sequence[Entity],
+        entities_to_keep: Sequence[Entity],
+        partial: bool,
+    ):
+        dynamodb = self._initialize_dynamodb()
+
+        for table_name in tables_to_keep:
+            table = None
+            try:
+                table = dynamodb.create_table(
+                    TableName=table_name.name,
+                    KeySchema=[
+                        {"AttributeName": "Row", "KeyType": "HASH"},
+                        {"AttributeName": "Project", "KeyType": "RANGE"},
+                    ],
+                    AttributeDefinitions=[
+                        {"AttributeName": "Row", "AttributeType": "S"},
+                        {"AttributeName": "Project", "AttributeType": "S"},
+                    ],
+                    ProvisionedThroughput={
+                        "ReadCapacityUnits": self._rcu,
+                        "WriteCapacityUnits": self._wcu,
+                    },
```

Comment:
I'd remove these and just use pay-per-request mode here so that users have fewer configuration options to care about.

We could leave provisioned throughput mode as an advanced option, but my hunch is that even this is a stretch. I suspect it would very rarely make sense cost wise, given how bursty Feast write workloads are.